File: bindings/typescript/examples/quickstart.ts
/**
 * Quickstart: drive the bridge over both transports.
 *
 *   LLMCORE_BRIDGE_FAKE=1 python -m llmcore.bridge.cli serve \
 *     --transport grpc,http --grpc-address 127.0.0.1:50151 \
 *     --http-address 127.0.0.1:50152 --insecure
 *
 *   npx tsx examples/quickstart.ts
 */
import { LlmcoreGrpcClient, LlmcoreHttpClient } from "../src/index";

const GRPC = process.env.LLMCORE_GRPC ?? "127.0.0.1:50151";
const HTTP = process.env.LLMCORE_HTTP ?? "http://127.0.0.1:50152";

async function main(): Promise<void> {
  // ---- gRPC ----
  const grpc = new LlmcoreGrpcClient(GRPC);
  const info = await grpc.ensureCompatible(["tier0"]);
  console.log(`[grpc] contract=${info.contractVersion} caps=${info.capabilities.join(",")}`);

  const chat = await grpc.chat({ message: "hello from grpc" });
  console.log(`[grpc] chat -> ${JSON.stringify(chat.text)} tokens=${chat.usage?.totalTokens}`);

  process.stdout.write("[grpc] stream -> ");
  for await (const chunk of grpc.chatStream({ message: "stream me" })) {
    if (!chunk.done) process.stdout.write(chunk.text);
  }
  process.stdout.write("\n");
  grpc.close();

  // ---- HTTP ----
  const http = new LlmcoreHttpClient(HTTP);
  await http.ensureCompatible();
  const hres = await http.chat({ message: "hello from http" });
  console.log(`[http] chat -> ${JSON.stringify(hres.text)}`);
  console.log(`[http] providers -> ${JSON.stringify((await http.listProviders()).providers)}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: bindings/c/examples/quickstart.c
/* Quickstart: drive the bridge over HTTP/SSE.
 *
 *   LLMCORE_BRIDGE_FAKE=1 python -m llmcore.bridge.cli serve \
 *     --transport http --http-address 127.0.0.1:50152 --insecure
 *   ./quickstart            # or set LLMCORE_HTTP=http://host:port
 */
#include <stdio.h>
#include <stdlib.h>

#include "llmcore_client.h"

static int print_chunk(const char *text, int done, void *user) {
  (void)user;
  if (!done) fputs(text, stdout);
  return 0; /* return nonzero to cancel */
}

int main(void) {
  const char *base = getenv("LLMCORE_HTTP");
  if (!base) base = "http://127.0.0.1:50152";

  llmcore_client *c = llmcore_client_new(base);
  if (!c) {
    fprintf(stderr, "client alloc failed\n");
    return 1;
  }

  const char *caps[] = {"tier0"};
  llmcore_error *e = llmcore_ensure_compatible(c, caps, 1);
  if (e) {
    fprintf(stderr, "incompatible: %s %s\n", e->category, e->message);
    llmcore_error_free(e);
    llmcore_client_free(c);
    return 1;
  }

  llmcore_chat_result r;
  e = llmcore_chat(c, "hello from c", &r);
  if (e) {
    fprintf(stderr, "chat error: %s %s\n", e->category, e->message);
    llmcore_error_free(e);
  } else {
    printf("chat -> %s (tokens=%d)\n", r.text, r.total_tokens);
    llmcore_chat_result_free(&r);
  }

  printf("stream -> ");
  fflush(stdout);
  e = llmcore_chat_stream(c, "stream me", print_chunk, NULL);
  printf("\n");
  if (e) {
    fprintf(stderr, "stream error: %s %s\n", e->category, e->message);
    llmcore_error_free(e);
  }

  llmcore_client_free(c);
  return 0;
}
